### PR TITLE
Initial rangeslider range

### PR DESCRIPTION
**Pull request by mdtusz**
@etpinard @jackparmer 

Adds a feature to rangesliders so you can specify an initial range.

In brief:

* Set `rangeslider.range` default - because the axis range is updated further down the codepath, we default to `[-Infinity, Infinity]` so as to use the full domain.
* If an axis range is specified, this will be the _full_ range of the slider.
* Make the call to set the range!
* Tests!

### CODE REVIEW

File: src/plot_api/plot_api.js [hunk 1]
```diff
@@ -310,6 +309,7 @@ Plotly.plot = function(gd, data, layout, config) {
         Shapes.drawAll(gd);
         Plotly.Annotations.drawAll(gd);
         Legend.draw(gd);
+        RangeSlider.draw(gd);
```
Comment (etpinard) [inline src/plot_api/plot_api.js]: All layout components are side-by-side. Looking good!

File: test/jasmine/tests/range_slider_test.js [hunk 2]
```diff
@@ -266,7 +269,8 @@ describe('the range slider', function() {
                     thickness: 'invalid',
                     bgcolor: 42,
                     bordercolor: 42,
-                    borderwidth: 'superfat'
+                    borderwidth: 'superfat',
+                    range: 'power-rangers-GO!'
```
Comment (etpinard) [inline test/jasmine/tests/range_slider_test.js]: :rocket:

File: src/components/rangeslider/attributes.js [hunk 3]
```diff
@@ -30,6 +30,22 @@ module.exports = {
...
     },
+    range: {
+        valType: 'info_array',
+        role: 'info',
+        items: [
+            {valType: 'number'},
+            {valType: 'number'}
+        ],
+        description: [
+            'Sets the initially selected range for the range slider.',
```
Comment (etpinard) [inline src/components/rangeslider/attributes.js]: @mdtusz Hmm. That's not how I imagined it.

I was thinking that the *initially selected range* would be given by `xaxis.range` - as done currently on first render as well as on range updates.

Instead, `xaxis.rangeslider.range` would correspond to the end-to-end range (or span) of the range slider.

For example,

```js
layout.xaxis = {
  range: [10, 20],
  rangeslider: {
    range: [0, 30]
  }
}
```

 would imply the the initially selected range is `[10, 20]` whereas the slider has a span of `[0, 30]`

image

Comment (mdtusz) [inline src/components/rangeslider/attributes.js]: Ahh I see - crossed wires on this one but that makes way more sense. I'll close this PR and open a new one once I've got it swapped about - shouldn't take too long.

**Pull request closed without merging**